gqcp/tests/ONVBasis/ONVPath_test.cpp

```diff
@@ -0,0 +1,153 @@
+
+    // A translation to the left means that we've encountered an electron, so the sign should be updated.
+    BOOST_REQUIRE(onv_path.sign() == 1);
+
+    // Since we've removed an arc with weight 1, and created one with weight 3, the address of the open path should now be 5.
+    BOOST_REQUIRE(onv_path.address() == 5);
+
+
+    // The next orbital that should be checked for creation should now have index 3, since we have moved up one orbital index.
+    BOOST_REQUIRE(onv_path.nextCreationOrbitalIndex() == 4);
+
+
+    // We can now close up the path by creating an electron in the orbital with index 4 (the current creation index).
+    onv_path.create(4, 2);
+
+    // Since we've created an arc with weight 4, the address should now be 9.
+    BOOST_REQUIRE(onv_path.address() == 9);
+
+    // The total sign factor of this path is 1, since we've encountered 2 electrons in total.
+    BOOST_REQUIRE(onv_path.sign() == 1);
+}
+
+/**
+ *  Test if the shift in address, orbital indices and sign change correspond to the correct solutions
+ *  
+ *  This method does not alter the ONVs in any way.
+ */
+BOOST_AUTO_TEST_CASE(leftTranslateUntilVertical) {
+
+    // Set up a F(5,3) Fock space.
+    const size_t M = 5;
+    const size_t N = 3;
+
+    const GQCP::SpinUnresolvedONVBasis onv_basis {M, N};
+
+
+    // Set up the reference values from an example. [LINK TO WEBSITE]
+    const auto I = GQCP::SpinUnresolvedONV::FromOccupiedIndices({0, 2, 3}, 5);  // |10110>
+    GQCP::ONVPath onv_path {onv_basis, I};
+    BOOST_REQUIRE(onv_path.address() == 2);
+
+    // We start our example by annihilating the first electron on orbital index 0. This creates an open path.
+    onv_path.annihilate(0, 0);
+
+    // Starting from the index of the removed electron, we shift the "next possible creation operator" until we find an unoccupied orbital. In our case, this should be at p=1.
+    onv_path.leftTranslateUntilVertical();
+    auto nextUnoccupiedOrbitalIndex = onv_path.nextCreationOrbitalIndex();
+    auto nextUnoccupiedElectronIndex = onv_path.nextCreationElectronIndex();
+    BOOST_REQUIRE(nextUnoccupiedOrbitalIndex == 1 && nextUnoccupiedElectronIndex == 0);
+    // Since there were no left translations in this example, the address stays the same.
+    BOOST_REQUIRE(onv_path.address() == 2);
+
+    // We create the first electron up to this orbital.
+    onv_path.create(nextUnoccupiedOrbitalIndex, nextUnoccupiedElectronIndex);
+
+    // Subsequently, the second electron at orbital index 3 is annihilated.
+    onv_path.annihilate(2, 1);
+
+    // Starting from the removed electron, next unocupied orbital should be located at p=4.
+    onv_path.leftTranslateUntilVertical();
+    nextUnoccupiedOrbitalIndex = onv_path.nextCreationOrbitalIndex();
+    nextUnoccupiedElectronIndex = onv_path.nextCreationElectronIndex();
+    BOOST_REQUIRE(nextUnoccupiedOrbitalIndex == 4 && nextUnoccupiedElectronIndex == 2);
+    BOOST_REQUIRE(onv_path.address() == 4);
```

Comment:
I'd like you to rewrite this test from the ground up, possibly splitting it up in a couple of cases.

You'll have to test
- [ ] `create()`
- [ ] `annihilate()`
- [ ] `leftTranslateUntilVertical()`

Reply:
I've added a separate test for `annihilate()`and `create()`functions. However, I'm not sure what to do with the `leftTranslateUntilVertical()` test.

Reply:
I reasonable unit test for this case would be to mimic the second part of the example page, the one where two left translations are necessary to find a vertical arc.